packages/terra-switch/src/Switch.jsx

```diff
@@ -0,0 +1,146 @@
+import React, { useRef } from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames/bind';
+import { KEY_SPACE, KEY_RETURN } from 'keycode-js';
+import styles from './Switch.module.scss';
+
+const cx = classNames.bind(styles);
+
+// Translations Required
+const SWITCH_STATE = Object.freeze({
+  ON: 'On',
+  OFF: 'Off',
+});
+
+const propTypes = {
+  /**
+   * Id of the Switch button.
+   */
+  id: PropTypes.string.isRequired,
+  /**
+   * Id of the Switch label.
+   */
+  labelId: PropTypes.string.isRequired,
+  /**
+   * Whether or not the Switch is checked ("ON").
+   */
+  checked: PropTypes.bool,
+  /**
+   * Whether or not the Switch is disabled.
+   */
+  disabled: PropTypes.bool,
+  /**
+   * The label of the Switch component.
+   */
+  label: PropTypes.node.isRequired,
+  /**
+   * Callback function when switch value changes from ON / OFF.
+   * Parameters: 1. switch value 2. event.
+   */
+  onChange: PropTypes.func,
+};
+
+const defaultProps = {
+  checked: false,
+  disabled: false,
+  onChange: undefined,
+};
+
+const Switch = (props) => {
+  const {
+    id,
+    labelId,
+    checked,
+    disabled,
+    onChange,
+    label,
+    ...customProps
+  } = props;
+
+  const sliderButton = useRef();
+
+  const handleClick = (event) => {
+    // See https://developer.mozilla.org/en-US/docs/Web/HTML/Element/Button#Clicking_and_focus
+    // Button on Firefox, Safari and IE running on OS X does not receive focus when clicked.
+    // This will set focus on the button when clicked.
+    sliderButton.current.focus();
+    if (onChange) {
+      onChange(!checked, event);
+    }
+  };
+
+  const handleOnKeyDown = (event) => {
+    if (event.keyCode === KEY_SPACE || event.keyCode === KEY_RETURN) {
+      event.preventDefault();
+    }
+  };
+
+  const handleKeyUp = (event) => {
+    if (event.keyCode === KEY_SPACE || event.keyCode === KEY_RETURN) {
+      if (onChange) {
+        onChange(!checked, event);
+      }
+    }
+  };
+
+  const switchClassNames = cx([
+    'switch',
+  ]);
+
+  const trayClassNames = cx([
+    'tray',
+    { selected: checked },
+    { unselected: !checked },
+    { 'is-disabled': disabled },
+  ]);
+
+  const sliderClassNames = cx([
+    'slider',
+    { 'is-disabled': disabled },
+  ]);
+
+  const statusLabelText = checked ? SWITCH_STATE.ON : SWITCH_STATE.OFF;
+
+  let mainClasses = cx('switch-wrapper');
+  if (customProps.className) {
+    mainClasses = `${mainClasses} ${customProps.className}`;
+  }
+
+  delete customProps.className;
+
+  return (
+    <div className={mainClasses} {...customProps}>
```

Comment:
```suggestion
    <div {...customProps} className={mainClasses}>
```
If you spread customProps first the className we supply will overwrite className provided in customProps if provided. So you wouldn't need to delete it above.

Reply:
I think deleting class-name here adds value for readability than overwriting it with placing first. so i will keep both, moved customProps to first [here](https://github.com/cerner/terra-core/pull/2846/commits/f4692ff31e65e7a746c1e364be845ab3210915a6#diff-ce03daf8e4362de3a438b2f322dedf6f)

Reply:
I agree with @jmsv6d, after referring a few terra-form components we **do not** need to delete the customProps classname for the sake of consistency among terra components

Reply:
As I mentioned  in above comment. I would like to keep it for readability so that it explicitly defines that classname prop in customProp is not considered when spread on component.